tests/files/tests.py

```diff
@@ -38,6 +40,23 @@ def test_context_manager(self):
         self.assertTrue(f.closed)
         self.assertTrue(orig_file.closed)
 
+    def test_open_resets_opened_file_to_start_and_returns_context_manager(self):
+        file = File(BytesIO(b'content'))
```

Comment:
In the first test, I'm using BytesIO because I want to test that seek(0) is done. We don't need an actual file on disk for this. See test_file_iteration().
OTOH, the second test does need an actual file because a stream cannot be closed/re-opened. See test_context_manager().